test/protobuf_converter_test.dart

```diff
@@ -183,6 +186,189 @@ void main() {
             result.iif(success: id, failure: (_) => throw Exception()), value);
       });
     });
+
+    group("ConnectionStateUpdate conversion", () {
+      const id = 'id';
+      const connectionState = 1;
+      final message = pb.DeviceInfo()
+        ..id = id
+        ..connectionState = connectionState;
+
+      test('Converts id', () {
+        final result = sut.connectionStateUpdateFrom(message);
+        expect(result.deviceId, id);
+      });
+
+      test('Converts status update', () {
+        final result = sut.connectionStateUpdateFrom(message);
+        expect(result.connectionState, DeviceConnectionState.connected);
+      });
+
+      test('converts failure', () {
+        final failure = pb.GenericFailure()
+          ..code = 0
+          ..message = "failure";
+        final failedUpdate = pb.DeviceInfo()..failure = failure;
+
+        final updateResult =
+            sut.connectionStateUpdateFrom(failedUpdate).failure;
```

Comment:
It's confusing that in most of the test cases in this group `sut.connectionStateUpdateFrom(·)` operates on `message`, but here the pattern breaks. I'd suggest separating the test-cases into a few groups so that the test-cases are as uniform in each one as possible.